src/SFML/Network/Ftp.cpp

```diff
@@ -269,43 +269,43 @@ Ftp::Response Ftp::deleteDirectory(const std::string& name)
 
 
 ////////////////////////////////////////////////////////////
-Ftp::Response Ftp::renameFile(const std::string& file, const std::string& newName)
+Ftp::Response Ftp::renameFile(const std::filesystem::path& file, const std::filesystem::path& newName)
 {
-    Response response = sendCommand("RNFR", file);
+    Response response = sendCommand("RNFR", file.string());
     if (response.isOk())
-       response = sendCommand("RNTO", newName);
+       response = sendCommand("RNTO", newName.string());
 
     return response;
 }
 
 
 ////////////////////////////////////////////////////////////
-Ftp::Response Ftp::deleteFile(const std::string& name)
+Ftp::Response Ftp::deleteFile(const std::filesystem::path& name)
 {
-    return sendCommand("DELE", name);
+    return sendCommand("DELE", name.string());
 }
 
 
 ////////////////////////////////////////////////////////////
-Ftp::Response Ftp::download(const std::string& remoteFile, const std::string& localPath, TransferMode mode)
+Ftp::Response Ftp::download(const std::filesystem::path& remoteFile, const std::filesystem::path& localPath, TransferMode mode)
 {
     // Open a data channel using the given transfer mode
     DataChannel data(*this);
     Response response = data.open(mode);
     if (response.isOk())
     {
         // Tell the server to start the transfer
-        response = sendCommand("RETR", remoteFile);
+        response = sendCommand("RETR", remoteFile.string());
         if (response.isOk())
         {
             // Extract the filename from the file path
-            std::string filename = remoteFile;
+            std::string filename = remoteFile.string();
```

Comment:
Since you're already introducing `<filesystem>` it would be better to use proper functions instead of only converting to string every time. `std::filesystem::path` is already implicitly convertible to `std::string` so only that change would not introduce any benefit at all:
```cpp
void usePath(const std::string& path);

usePath(std::filesystem::current_path());  // works
```

Here the comment says: `// Extract the filename from the file path` so maybe the intention was `remoteFile.filename()` or `remoteFile.stem()`

Reply:
`std::filesystem::path` is convertible to `std::string` on Linux and macOS but not on Windows. I had to add `.string()` all over the place to appease the Windows CI jobs that kept failing without that. Maybe we should using `auto` in more contexts so Windows builds can automatically use wide character strings but other OSes can use `std::string`.

Reply:
> it would be better to use proper functions instead of only converting to string every time
You're right. I was just too lazy to replace this code without whatever the `<filesystem>` counterpart is. I just updated the PR doing exactly that. It's a lot shorter and easier to read now.